src/components/EventList.tsx

```diff
@@ -60,15 +65,27 @@ export default function EventList ({ eventResponses, events, onSignup, onUndoSig
 }
 
 interface EventListItemProps {
+    booked: BookedEvent | undefined;
     event: ZetkinEvent;
-    org: ZetkinOrganization;
-    response: ZetkinEventResponse | undefined;
     onSignup: (eventId: number, orgId: number) => void;
     onUndoSignup: (eventId: number, orgId: number) => void;
+    org?: ZetkinOrganization;
+    response: ZetkinEventResponse | undefined;
 }
 
-const EventListItem = ({ event, response, onSignup, onUndoSignup, org }: EventListItemProps): JSX.Element => {
+const EventListItem = ({ booked, event, response, onSignup, onUndoSignup, org }: EventListItemProps): JSX.Element => {
     const user = useUser();
+    let todo : boolean;
+
+    if (!org) {
+        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+        org = event.organization!;
+        todo = true;
+    }
+    else {
+        todo = false;
+    }
+
```

Comment:
What is the purpose of this `todo` flag? As far as I can tell from this logic, it's used to derive whether the component is being rendered in the context of the todo page (where per-event organizations exist) or not? That's a code smell.

A component should generally not know on what page it's being rendered.